opencti-platform/opencti-graphql/src/migrations/1688710489709-add_order_to_opinion_ov.js

```diff
@@ -0,0 +1,63 @@
+import { Promise } from 'bluebird';
+import { executionContext, SYSTEM_USER } from '../utils/access';
+import { logApp } from '../config/conf';
+import { elRawUpdateByQuery, elReplace, ES_MAX_CONCURRENCY } from '../database/engine';
+import { READ_INDEX_STIX_META_OBJECTS } from '../database/utils';
+import { openVocabularies } from '../modules/vocabulary/vocabulary-utils';
+import { VocabularyCategory, VocabularyFilter } from '../generated/graphql';
+import { listAllEntities } from '../database/middleware-loader';
+import { ENTITY_TYPE_VOCABULARY } from '../modules/vocabulary/vocabulary-types';
+import { DatabaseError } from '../config/errors';
+
+export const up = async (next) => {
+  const context = executionContext('migration');
+  const start = new Date().getTime();
+  logApp.info('[MIGRATION] Adding default order value to opinion open vocabulary');
+
+  const updateEntityQuery = {
+    script: {
+      params: { order: 0 },
+      source: 'ctx._source.order = params.order;',
+    },
+    query: {
+      term: { 'entity_type.keyword': { value: ENTITY_TYPE_VOCABULARY } }
+    },
+  };
+  await elRawUpdateByQuery({
+    index: [READ_INDEX_STIX_META_OBJECTS],
+    refresh: true,
+    wait_for_completion: true,
+    body: updateEntityQuery
+  })
+    .catch((err) => {
+      throw DatabaseError('Error updating elastic', { error: err });
+    });
+
+  const defaultVocabularies = new Map((openVocabularies.opinion_ov ?? []).map((v) => [v.key, v]));
+  const filters = [{
+    key: [VocabularyFilter.Category],
+    values: [VocabularyCategory.OpinionOv]
+  }];
+  const vocabularies = await listAllEntities(context, SYSTEM_USER, [ENTITY_TYPE_VOCABULARY], {
+    indices: [READ_INDEX_STIX_META_OBJECTS],
+    connectionFormat: false,
+    filters,
+  });
+
+  const updateVocabulary = async (vocabulary) => {
+    const defaultVocabulary = defaultVocabularies.get(vocabulary.name);
+    if (defaultVocabulary) {
+      const patch = { order: defaultVocabulary.order };
+      await elReplace(vocabulary._index, vocabulary.id, { doc: patch });
+    }
+  };
+
+  await Promise.map(vocabularies, updateVocabulary, { concurrency: ES_MAX_CONCURRENCY });
```

Comment:
what is the difference between updateEntityQuery and updateVocabulary ? Why do we need both for the migration ?

Reply:
Would it work if we get the defaultVocabularies and use them as a param in the script above ? then we set the order with either the default vocabulary order, or 0

Reply:
We need to add, by default, 0 to all existing vocabulary.
Add for opinion, we need to have a valid order to display opinion in radar correctly.

Reply:
Ok I understand now, still I think we could do everything in the same script, something like :

```
script: {
      params: { defaultVocabulariesList },
      source: `ctx._source.order = 0; 
          if (entity.category=='opinion_ov') { 
             for (defaultVocab in params.defaultVocabularies) { 
                 if (defaultVocab.name.equals(entity.name)) {
                    entity.order = defaultVocab.order;
                 } 
             } 
         }`,
}
```

But maybe it's not really important, if the migration works already, I guess it's not necessary to change it.